src/main/java/com/example/bookshop/controllers/BookController.java

```diff
@@ -14,36 +18,48 @@
 import org.springframework.web.bind.annotation.PutMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
+@Tag(name = "Book management", description = "Endpoints for manage books")
 @RestController
 @RequestMapping(value = "/books")
 public class BookController {
     @Autowired
     private BookService bookService;
 
+    @GetMapping("/search")
+    public List<BookDto> getByCriteria(@RequestParam Map<String, List<String>> params) {
```

Comment:
create BookSearchParametersDto; consufing name `getByCriteria`
```suggestion
    public List<BookDto> searchBooks(BookSearchParametersDto searchParameters) {
```